app/lib/identity_tijuana/member_sync.rb

```diff
@@ -109,15 +109,20 @@ def self.get_id_change_date(member, field_info_type, default_change_date = nil)
       end
       # Iterate through the relevant audit log list, moving backwards in time
       # until we find a change to the field or fields in question.
-      member.send(audit_list_name).where(auditable_type: auditable_type).reverse_each do |audit|
+      # Note that we use `reorder` to remove all prior ordering clauses,
+      # as Audited injectes `order by version asc` when order clause is used.
+      member.send(audit_list_name)
+            .where(auditable_type: auditable_type)
+            .reorder(created_at: :desc)
+            .each do |audit|
```

Comment:
Can you walk me through why the versioning is not reliable but the created_at is?

Reply:
> Can you walk me through why the versioning is not reliable but the created_at is?

`member.send(audit_list_name).where(auditable_type: auditable_type).reverse_each` query will not apply any ordering, and the resulting set is arbitrarily ordered.

The `ORDER BY version ASC` is injected as the primary order clause only when `order` method is called. In essence, yes we can call `order(version: :desc)` and we'll get the same ordering.

I've opted for `reorder` with `created_at` as it is indexed column, as opposed to `version`, but it also made more sense to keep aligned with `moving backwards in time` in the iteration through the relevant audit logs.

Reply:
Got it, thanks. Can you update the comment to say something similar?

It's clear from the code and the `reorder` method's API docs that the reordering is happening and what its effects are, but it's not clear why that is needed.